rllib/examples/documentation/checkpoints_and_exports.py

```diff
@@ -0,0 +1,261 @@
+# Note that for normal, single-agent Algorithms, the Policy ID is "default_policy".
+policy1 = my_ma_algo.get_policy(policy_id="pol1")
+
+# Tell RLlib to store an individual policy checkpoint (only for "pol1") inside
+# /tmp/my_policy_checkpoint
+policy1.export_checkpoint("/tmp/my_policy_checkpoint")
+
+# __create-policy-checkpoint-end__
+
+# __restore-policy-begin__
+
+import numpy as np  # noqa
+
+from ray.rllib.policy.policy import Policy  # noqa
+
+# Use the `from_checkpoint` utility of the Policy class:
+my_restored_policy = Policy.from_checkpoint("/tmp/my_policy_checkpoint")
+
+# Use the restored policy for serving actions.
+obs = np.array([0.0, 0.1, 0.2, 0.3])  # individual CartPole observation
+action = my_restored_policy.compute_single_action(obs)
+
+print(f"Computed action {action} from given CartPole observation.")
+
+# __restore-policy-end__
+
+
+# __restore-algorithm-from-checkpoint-with-fewer-policies-begin__
+
+from ray.rllib.algorithms.ppo import PPOConfig  # noqa
+from ray.rllib.examples.env.multi_agent import MultiAgentCartPole  # noqa
+
+# Set up an Algorithm with 5 Policies.
+algo_w_5_policies = (
+    PPOConfig()
+    .environment(
+        env=MultiAgentCartPole,
+        env_config={
+            "num_agents": 5,
+        },
+    )
+    .multi_agent(
+        policies={"pol0", "pol1", "pol2", "pol3", "pol4"},
+        # Map "agent0" -> "pol0", etc...
+        policy_mapping_fn=(
+            lambda agent_id, episode, worker, **kwargs: f"pol{agent_id}"
+        ),
+    )
+    .build()
+)
+
+# .. train one iteration ..
+algo_w_5_policies.train()
+# .. and call `save()` to create a checkpoint.
+path_to_checkpoint = algo_w_5_policies.save()
+print(
+    "An Algorithm checkpoint has been created inside directory: "
+    f"'{path_to_checkpoint}'. It should contain 5 policies in the 'policies/' sub dir."
+)
+# Let's terminate the algo for demonstration purposes.
+algo_w_5_policies.stop()
+
+# We will now recreate a new algo from this checkpoint, but only with 2 of the
+# original policies ("pol0" and "pol1"). Note that this will require us to change the
+# `policy_mapping_fn` (instead of mapping 5 agents to 5 policies, we now have
+# to map 5 agents to only 2 policies).
+
+
+def new_policy_mapping_fn(agent_id, episode, worker, **kwargs):
+    return "pol0" if agent_id in ["agent0", "agent1"] else "pol1"
+
+
+algo_w_2_policies = Algorithm.from_checkpoint(
+    checkpoint=path_to_checkpoint,
+    policies={"pol0", "pol1"},  # <- restore only those policy IDs here.
+    policy_mapping_fn=new_policy_mapping_fn,  # <- use this new mapping fn.
+)
+
+# Test, whether we can train with this new setup.
+algo_w_2_policies.train()
+# Terminate the new algo.
+algo_w_2_policies.stop()
+
+# __restore-algorithm-from-checkpoint-with-fewer-policies-end__
+
+
+# __export-models-begin__
+
+from ray.rllib.algorithms.ppo import PPOConfig  # noqa
+
+# Create a new Algorithm (which contains a Policy, which contains a NN Model).
```

Comment:
This whole import is too long and consists of 80%+ comments. I think this doesn't read very well on the docs (as a reader I'm not sure if I'm supposed to read all this, it looks as if all those comments are there by "accident"):

![Screenshot 2022-10-07 at 13 23 55](https://user-images.githubusercontent.com/3462566/194542966-5696d95e-86fe-4d49-8044-70092c50f120.png)

Reply:
basically this import and maybe the last one suffer from this a little, other than that I think this looks great btw.

Reply:
I'll split it up and move the comments into the rsv file.

Reply:
Left the keras-related comments in the code, but split up the 3 different ways on how to save your models (direct, via policy checkpoint, via algo checkpoint).